eslint.config.mjs

```diff
@@ -0,0 +1,155 @@
+import base, { createConfig } from '@metamask/eslint-config';
```

Comment:
ESLint configuration using the new [flat config format](https://eslint.org/docs/latest/use/configure/migration-guide)
Key differences from old .eslintrc format:
- Config is an array of objects, each acting like an "override" 
- Plugins/parsers are imported directly as modules
- Files patterns must be explicitly specified(we differentiate from core because we use `tsx` as well)
- All paths are relative to config file location
- No more "extends" - configs are composed using array spread